src/hooks/useAffiliatesInfo.ts

```diff
@@ -38,11 +50,31 @@ export const useAffiliatesInfo = (dydxAddress?: string) => {
     }
   };
 
-  const query = useQuery({
+  const affiliateMetadataQuery = useQuery({
     queryKey: ['affiliatesMetadata', dydxAddress],
-    queryFn,
+    queryFn: fetchAffiliateMetadata,
     enabled: Boolean(compositeClient && dydxAddress),
   });
 
-  return query;
+  const fetchAffiliateMaxEarning = async () => {
+    const taker3FeeTier = feeTiers?.[2].taker ?? DEFAULT_TAKER_3_FEE;
+    const allAffiliateTiers = await getAllAffiliateTiers();
+    let maxRevshare = DEFAULT_MAX_AFFILIATE_SHARE;
+    if (allAffiliateTiers?.tiers?.tiers?.length) {
+      const lastTier = allAffiliateTiers.tiers.tiers[allAffiliateTiers.tiers.tiers.length - 1];
```

Comment:
[non-blocking] if this is a regular js array i think we can use `allAffiliateTiers.tiers.tiers.at(-1)` to get the last time, which is convenient